### PR TITLE
all errors except for contact info california

### PR DESCRIPTION
**Pull request by ashleynguyen27**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?
In this PR, I am adding errors based on Iris's designs to onboarding and the login page.
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### How to Review
This PR only includes things related to onboarding and the login page. Check out these files:

onboardinUtils.js - most error message updates
Login.js - added error message for invalid login credentials 
Onboarding.js - onboarding general function added 
* BasicInfoStep.js - subpage of onboarding 
* ContactInfoStep.js - ^
* PaymentDetailsStep.js - ^
various css files (Login, Onboarding, PPModal, main) 

to test go through onboarding and intentionally make mistakes
[//]: # 'The order in which to review files and 
what to expect when testing locally'
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Related PRs

[//]: # "Optional - related PRs you're waiting on
/ PRs that will conflict, etc"
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Next steps

[//]: # "What doesn't work yet, what's NOT in this 
PR/has to be done "
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Tests Performed, Edge Cases

[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Screenshots

[//]: # '#####################################'
[//]: # '#### YOUR SCREENSHOTS BELOW ########'
[//]: # '############## END ##################'

CC: @dfangshuo

### CODE REVIEW

File: src/styles/PPModal.css [hunk 1]
```diff
@@ -50,17 +50,27 @@
   font-size: 16px;
   line-height: 22px;
   margin-bottom: 2rem;
-  word-break: break-all;
+  word-break: normal;
```
Comment (dfangshuo) [inline src/styles/PPModal.css]: im not sure of the answer to this either, but does this need to be specified if the behavior is normal? Or can we just delete this line altogether and the default behavior would be 'normal'?

File: src/styles/Login.css [hunk 2]
```diff
@@ -65,3 +65,21 @@ w-100 {
...
+  margin-right: auto;
+}
+
+.mt-15 {
+  margin-top: 1.5rem;
+}
+.error-icon {
+  margin-right: 1rem;
+}
+
```
Comment (dfangshuo) [inline src/styles/Login.css]: nit: not sure if u wanna name the bottom .mr-15 or something just to be consistent

File: src/lib/onboardingUtils.js [hunk 3]
```diff
@@ -7,12 +8,28 @@ import {
   createOwner
 } from './airtable/request';
 import { refreshUserData } from './userDataUtils';
+import Error from '../assets/error.svg';
```
Comment (dfangshuo) [inline src/lib/onboardingUtils.js]: Name this ErrorImage or something to prevent confusion

File: src/lib/onboardingUtils.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 /* eslint-disable no-await-in-loop */
+import React from 'react';
 import States from '../assets/states.json';
```
Comment (dfangshuo) [inline src/lib/onboardingUtils.js]: maybe rename this USStates or something to prevent confusion. Rename both the json files and the variable name

File: src/lib/onboardingUtils.js [hunk 5]
```diff
@@ -53,9 +72,14 @@ const validateShares = value => {
 
 // Ensure State is a real state (either abbreivation or full name)
 const validateState = value => {
-  return States.map(s => s.toUpperCase()).indexOf(value.toUpperCase()) !== -1
-    ? ''
-    : 'Invalid State';
+  const upperCaseValue = value.toUpperCase();
+  if (States.map(s => s.toUpperCase()).indexOf(upperCaseValue) !== -1) {
```
Comment (dfangshuo) [inline src/lib/onboardingUtils.js]: im not exactly sure what this line is supposed to check for, is there a reason we can't just perform the inner if check?

File: src/lib/onboardingUtils.js [hunk 6]
```diff
@@ -53,9 +72,14 @@ const validateShares = value => {
 
 // Ensure State is a real state (either abbreivation or full name)
 const validateState = value => {
```
Comment (dfangshuo) [inline src/lib/onboardingUtils.js]: Rename this to ValidateCountryState or ValidateUSState or something to avoid confusion

File: src/screens/auth/Login.js [hunk 7]
```diff
@@ -3,13 +3,15 @@ import { loginUser } from '../../lib/authUtils';
 import '../../styles/Login.css';
 import '../../styles/main.css';
 import Constants from '../../constants';
+import Error from '../../assets/error.svg';
```
Comment (dfangshuo) [inline src/screens/auth/Login.js]: same comment as above about renaming this file name

File: src/screens/onboarding/Onboarding.js [hunk 8]
```diff
@@ -22,7 +22,7 @@ class Onboarding extends React.Component {
         onboardingStep: 0,
         inviteToken: '',
         ownerTypes: [GENERAL_OWNER],
-        isReceivingDividends: true,
+        isReceivingDividends: false,
```
Comment (dfangshuo) [inline src/screens/onboarding/Onboarding.js]: what's the reason for this change?

File: src/screens/onboarding/Onboarding.js [hunk 9]
```diff
@@ -40,6 +40,15 @@ class Onboarding extends React.Component {
...
 
+  checkValid = (input, type) => {
+    if (type === 0) {
+      return input !== '' && typeof input !== 'undefined'
+        ? 'b-is-not-valid'
+        : 'b-is-valid';
+    }
+    return !input ? '\u00A0' : input;
+  };
+
```
Comment (dfangshuo) [inline src/screens/onboarding/Onboarding.js]: are  we using 0 and non-0 to denote the 2 cases here?

File: src/screens/onboarding/steps/BasicInfoStep.js [hunk 10]
```diff
@@ -25,30 +25,26 @@ class BasicInfoStep extends React.PureComponent {
...
               defaultValue={owner.firstName}
-              className={` input-gray ${
-                errors.firstName !== '' ? 'b-is-not-valid' : 'b-is-invalid'
-              }`}
+              className={` input-gray ${checkValid(errors.firstName, 0)}`}
             />
           </div>
           <div className="w-50">
-            <div style={{ marginBottom: '20px' }} />
+            <div style={{ marginBottom: '22px' }} />
```
Comment (dfangshuo) [inline src/screens/onboarding/steps/BasicInfoStep.js]: nit: make a css class and style by using className

File: src/screens/onboarding/steps/PaymentDetailsStep.js [hunk 11]
```diff
@@ -1,8 +1,31 @@
 import React from 'react';
+import Error from '../../../assets/error.svg';
```
Comment (dfangshuo) [inline src/screens/onboarding/steps/PaymentDetailsStep.js]: similar comment as above

File: src/lib/onboardingUtils.js (hunk 5 above)
Comment (aivantg) [reply src/lib/onboardingUtils.js]: that's the "except california" part of this PR I think. This first IF is just to check valid state. We can change later

File: src/lib/onboardingUtils.js [hunk 12]
```diff
@@ -7,12 +8,28 @@ import {
...
 // Allows custom error message
-const validateExistence = (value, error = 'Required') => {
+const validateExistence = (
+  value,
+  error = 'Please enter this required field.'
+) => {
+  return value ? '' : error;
+};
+
+const errorText = (
```
Comment (aivantg) [inline src/lib/onboardingUtils.js]: Is this...the only error text that has complicated HTML?  Or can this be generalized to more errors

File: src/screens/auth/Login.js [hunk 13]
```diff
@@ -35,13 +37,20 @@ class Login extends React.Component {
       if (res.found && res.match) {
         this.segueToHome(evt);
       } else {
-        // alert('Invalid email or password!');
+        this.setState({
+          isValid: false
```
Comment (aivantg) [inline src/screens/auth/Login.js]: can we name this state variable something more specific, like "showLoginError" or something?  Might have to flip the true and false values there

File: src/screens/onboarding/Onboarding.js (hunk 8 above)
Comment (aivantg) [reply src/screens/onboarding/Onboarding.js]: yeah arguably we want the default to be "true"

File: src/screens/onboarding/steps/ContactInfoStep.js [hunk 14]
```diff
@@ -1,29 +1,61 @@
...
+
+  handleCloseModal() {
+    this.setState({ showModal: false });
+    window.location.reload();
+  }
+
+  returnToHomepage() {
+    const { owner } = this.props;
+    console.log(owner);
+    deleteRecord('Owner', owner.id);
```
Comment (aivantg) [inline src/screens/onboarding/steps/ContactInfoStep.js]: don't call `airtable.js` functions directly!! There is a function called `deleteOwnerRecord` in `request.js` you should import instead

File: src/screens/onboarding/steps/ContactInfoStep.js [hunk 15]
```diff
@@ -1,29 +1,61 @@
...
+      this.setState({ showModal: true });
+    }
+  }
+
+  handleCloseModal() {
+    this.setState({ showModal: false });
+    window.location.reload();
+  }
+
+  returnToHomepage() {
```
Comment (aivantg) [inline src/screens/onboarding/steps/ContactInfoStep.js]: Move this to `OnboardingUtils.js`

File: src/screens/onboarding/Onboarding.js (hunk 8 above)
Comment (ashleynguyen27) [reply src/screens/onboarding/Onboarding.js]: by default, dividend is checked to 'Yes'  but Iris's design has an error message if there not an option checked. So I thought it would make sense if none of the options were pre marked

File: src/screens/onboarding/Onboarding.js (hunk 9 above)
Comment (ashleynguyen27) [reply src/screens/onboarding/Onboarding.js]: yes

Comment (ashleynguyen27) [reply src/screens/onboarding/Onboarding.js]: oh im going to change if (!type)

File: src/lib/onboardingUtils.js (hunk 5 above)
Comment (aivantg) [reply src/lib/onboardingUtils.js]: oh jk, yeah this if statement is kind of a remnant of the old check

File: src/screens/onboarding/Onboarding.js [hunk 16]
```diff
@@ -40,6 +40,15 @@ class Onboarding extends React.Component {
     }
   }
 
+  checkValid = (input, type) => {
```
Comment (aivantg) [inline src/screens/onboarding/Onboarding.js]: What does this function do and what purpose does it serve :o

Comment (aivantg) [reply src/screens/onboarding/Onboarding.js]: it clearly works but i have no idea what it is LOL

File: src/screens/onboarding/Onboarding.js (hunk 8 above)
Comment (aivantg) [reply src/screens/onboarding/Onboarding.js]: hmm...I think it would make the dividend code less complicated if we just didn't allow for that error to ever happen. Like always having it checked to True? We can talk about this with @iris-hou  later though

File: src/screens/onboarding/Onboarding.js (hunk 16 above)
Comment (ashleynguyen27) [reply src/screens/onboarding/Onboarding.js]: i renamed it to checkValidCSSClass. basically sets the classname based on error validation

File: src/screens/onboarding/Onboarding.js (hunk 8 above)
Comment (ashleynguyen27) [reply src/screens/onboarding/Onboarding.js]: wait i dont think ^ is affected by this will change it back.

File: src/screens/auth/Login.js [hunk 17]
```diff
@@ -38,17 +38,17 @@ class Login extends React.Component {
...
       }
     } catch (err) {
       console.error(err);
     }
   };
 
-  checkValid() {
-    const { isValid } = this.state;
-    return isValid ? 'b-is-valid' : 'b-is-not-valid';
+  checkValidCSSClass() {
```
Comment (dfangshuo) [inline src/screens/auth/Login.js]: nit: i would replace CSSClass with Input or something

File: src/screens/onboarding/steps/ContactInfoStep.js (hunk 14 above)
Comment (dfangshuo) [reply src/screens/onboarding/steps/ContactInfoStep.js]: good catch. i definitely mised this lol

File: src/screens/auth/Login.js (hunk 17 above)
Comment (dfangshuo) [reply src/screens/auth/Login.js]: like 1toggleValidColor` or something more descriptive

File: src/screens/onboarding/Onboarding.js [hunk 18]
```diff
@@ -40,6 +40,15 @@ class Onboarding extends React.Component {
     }
   }
 
+  checkValidCSSClass = (input, type) => {
```
Comment (aivantg) [inline src/screens/onboarding/Onboarding.js]: could this be moved to onboardingUtils? Since it has nothing to do with state? That way it doesn't have to be passed down to every component.
